common/ManagementTestShared/MgmtRecordTestBase.cs

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using Azure.Core;
+using Azure.Core.TestFramework;
+#if RESOURCES_RP
+using Azure.ResourceManager.Resources;
+#else
+using Azure.Management.Resources;
+#endif
+using System;
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+
+namespace Azure.ResourceManager.TestFramework
+{
+    [Category("Recorded")]
+    public abstract class MgmtRecordTestBase<TEnvironment> : ClientTestBase where TEnvironment : TestEnvironment, new()
+    {
+        protected RecordedTestSanitizer Sanitizer { get; set; }
+
+        protected RecordMatcher Matcher { get; set; }
+
+        public TestRecording Recording { get; private set; }
+
+        public RecordedTestMode Mode { get; set; }
+
+        public TEnvironment TestEnvironment { get; }
+
+        private ResourceGroupCleanupPolicy CleanupPolicy { get; set; }
+
+        // copied the Windows version https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs
+        // as it is the most restrictive of all platforms
+        private static readonly HashSet<char> s_invalidChars = new HashSet<char>(new char[]
+        {
+            '\"', '<', '>', '|', '\0',
+            (char)1, (char)2, (char)3, (char)4, (char)5, (char)6, (char)7, (char)8, (char)9, (char)10,
+            (char)11, (char)12, (char)13, (char)14, (char)15, (char)16, (char)17, (char)18, (char)19, (char)20,
+            (char)21, (char)22, (char)23, (char)24, (char)25, (char)26, (char)27, (char)28, (char)29, (char)30,
+            (char)31, ':', '*', '?', '\\', '/'
+        });
+
+#if DEBUG
+        /// <summary>
+        /// Flag you can (temporarily) enable to save failed test recordings
+        /// and debug/re-run at the point of failure without re-running
+        /// potentially lengthy live tests.  This should never be checked in
+        /// and will be compiled out of release builds to help make that easier
+        /// to spot.
+        /// </summary>
+        public bool SaveDebugRecordingsOnFailure { get; set; } = false;
+#endif
+
+        protected MgmtRecordTestBase(bool isAsync) : this(isAsync, RecordedTestUtilities.GetModeFromEnvironment())
+        {
+        }
+
+        protected MgmtRecordTestBase(bool isAsync, RecordedTestMode mode) : base(isAsync)
+        {
+            Sanitizer = new RecordedTestSanitizer();
+            Matcher = new RecordMatcher();
+            Mode = mode;
+            TestEnvironment = new TEnvironment();
+            TestEnvironment.Mode = Mode;
+            CleanupPolicy = new ResourceGroupCleanupPolicy();
+        }
+
+        private string GetSessionFilePath()
```

Comment:
m-nash:
Looks like there are quite a few methods like this that are exact copies in RecordedTestBase, should we consider sharing these via a different inheritance heirarchy?

Reply:
@m-nash 
As I understand, @allenjzhang said this will replace the duplicates in RecordedTestBase by eventually deprecating that file and removing it. Although, since we will be making a mock version of this, yeah I agree we should put this in a common file.

Reply:
Why are we planning to remove `RecordedTestBase`?